vendor/bat-native-ledger/src/bat/ledger/internal/publisher/publisher.cc

```diff
@@ -257,7 +257,9 @@ void Publisher::SaveVisitInternal(
         publisher_info->favicon_url = fav_icon;
     }
   } else {
-    publisher_info->favicon_url = ledger::kClearFavicon;
+    if (publisher_info->favicon_url.empty()) {
+      publisher_info->favicon_url = ledger::kClearFavicon;
+    }
```

Comment:
During a page load, we save the pub info a lot. At one point during the page load/parse, we have a valid favicon url, at a later point, it is empty as it cannot be parsed. If an empty URL is passed in and we already have a favicon, this change prevents us clearing the existing favicon url. Can anyone see a problem with this, as other paths than Vimeo will use the same logic flow...

Reply:
I think we would have a problem with this code as if we save some wrong url that is not stored in chromium favicon storage then we will never clear it

Reply:
Bear in mind if we _do_ parse a URL, it will _always_ be saved. Also - if we implement the favicon cache expiration we mentioned, it will eventually self-repair even if the above did happen. WDYT? (I've yet to test this against production BTW - the change is here for discussion only ATM)

Reply:
Further - this doesn't seem to work anyway 🤔 which is a bit odd. The challenge ATM is that during the page scrape, the content doesn't have anything containing the Users portrait ID, so it ends up removing any prior Favicon.